src/settings.c

```diff
@@ -171,14 +177,19 @@ int git_libgit2_opts(int key, ...)
 		{
 			const char *file = va_arg(ap, const char *);
 			const char *path = va_arg(ap, const char *);
-			if (!SSL_CTX_load_verify_locations(git__ssl_ctx, file, path)) {
-				giterr_set(GITERR_NET, "SSL error: %s",
-					ERR_error_string(ERR_get_error(), NULL));
-				error = -1;
-			}
+			error = git_openssl_set_cert_file(file, path);
+		}
+#elif GIT_MBEDTLS
+		{
+			const char *file = va_arg(ap, const char *);
+			const char *path = va_arg(ap, const char *);
+			if (file)
+				error = git_mbedtls_set_cert_file(file, 0);
+			if (error && path)
+				error = git_mbedtls_set_cert_file(path, 0);
```

Comment:
I'm a bit undecided on this one. I guess how @omus proposes is the most sensible way, e.g. returning an error if neither file nor path are given. But what is unfortunate about this API is how we handle the case where both parameters are set, as only the file setting would be applied in that case. This does not really follow the way of least astonishment for the developer. On the other hand, we shouldn't throw an error in the case where both have been set, as otherwise developers would need to be aware whether libgit2 is compiled against mbedTLS or OpenSSL due to the latter accepting both at the same time.

I think the most sensible thing to do here is to use both values, file and path, in case both are set. Can this be supported in some way with mbedTLS?